File: automation-test-playwright/pages/core/CrudDetailPage.ts
import { type Locator, type Page } from "@playwright/test";
import { BasePage } from "./BasePage";

export class CrudDetailPage extends BasePage {
  protected readonly detailPath?: string;
  readonly pageHeader: Locator;

  /**
   * Initializes this page object.
   */
  constructor(page: Page) {
    super(page);
    this.pageHeader = this.firstVisible(page.locator("h1, h2"));
  }

  /**
   * Opens the page.
   */
  async open(id: number): Promise<void> {
    if (!this.detailPath) {
      throw new Error("This detail page does not define a detail path.");
    }

    await this.visit(`${this.detailPath}/${id}`);
  }

  async waitForHeaderContains(text: string): Promise<void> {
    await this.waitForLocatorText(this.pageHeader, text);
  }
}

File: automation-test-selenium/src/main/java/org/suntower/pages/core/CrudDetailPage.java
package org.suntower.pages.core;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.suntower.core.BasePage;

public class CrudDetailPage extends BasePage {
  protected String detailPath;
  protected final By pageHeader = css("h1, h2");

  public CrudDetailPage(WebDriver driver) {
    super(driver);
  }

  public void open(int id) {
    if (detailPath == null || detailPath.isBlank()) {
      throw new IllegalStateException("This detail page does not define a detail path.");
    }
    visit(detailPath + "/" + id);
  }

  public void waitForHeaderContains(String text) {
    waitForText(pageHeader, text);
  }
}
